cmd/storage-datatypes.go

```diff
@@ -490,8 +493,14 @@ type WriteAllHandlerParams struct {
 }
 
 // RenameDataResp - RenameData()'s response.
+// Provides information about the final state of Rename()
+//   - on xl.meta (array of versions) on disk to check for version disparity
+//   - on rewrite dataDir on disk that must be additionally purged
+//     only after as a 2-phase call, allowing the older dataDir to
+//     hang-around in-case we need some form of recovery.
 type RenameDataResp struct {
-	Signature uint64 `msg:"sig"`
+	Sign       []byte
```

Comment:
You should add a new `HandlerID` since the response isn't backwards compatible. `HandlerRenameData2` for example.

This will make `HandlerRenameData` calls from previous versions fail and `HandlerRenameData2` fail on newer versions in a mixed env.

Otherwise you will just get the zero value back from there.